analyzer/uncategorized/addresses.go

```diff
@@ -42,6 +45,15 @@ func StringifyEthAddress(ethAddr []byte) (apiTypes.Address, error) {
 	return StringifyOcAddress(ocAddr)
 }
 
+func EthAddrReference(preimage string) *string {
+	if len(preimage) == 0 {
```

Comment:
Does this ever happen? Feels like it shouldn't.

Reply:
Does :(

```
{
  "amount": "715000000000000000000",
  "body": {
    "amount": [
      "JsKeR8SETAAA",
      ""
    ],
    "to": "ABSac93FbIpG2PVK4dp6tXP4tkWw"
  },
  "fee": "2000000000000000",
  "gas_limit": 15000,
  "hash": "be42e50e8b65ce95ad6677c429659d7a6133c4e9e786e5cf8cd38be161ad7b37",
  "index": 1,
  "method": "consensus.Withdraw",
  "nonce_0": 140,
  "round": 1009434,
  "sender_0": "oasis1qqa20swrvp8hl4fz6cc2c5g7dq5eeuz28g00hf3j",
  "sender_0_eth": "0xAc09Ca840c02e9883fEB40A95eC7f7E506799601",
  "success": true,
  "timestamp": "2022-04-11T19:00:02-05:00",
  "to": "oasis1qq2f5u7ac4kg53kc749wrkn6k4el3dj9kqeq85ek"
}
```

We don't have a preimage for `oasis1qq2f5u7ac4kg53kc749wrkn6k4el3dj9kqeq85ek`.

Uh, alright, I'll dig a little more and see if we missed anything in prework.

Reply:
It's OK to not have preimages for some addrs. We only learn the preimage when the account actively submits a tx. So if somebody sends tokens _to_ an address that's never submitted anything before ... we have the above situation.

I assumed that for such addresses, we'd never reach `EthAddrReference()` in Go, but I reread now and I see I was wrong.

Reply:
https://github.com/oasisprotocol/oasis-indexer/blob/xz/3u9y3yp-add-transaction-sender-eth-address/storage/migrations/02_oasis_3_runtimes.up.sql#L252

although wouldn't this reference constraint have prevented us from scanning that block in the first place?

Reply:
> wouldn't this reference constraint have prevented us from scanning that block in the first place?

The constraint is only on the tx sender. Xi's example has a `to` with an unknown eth address, not a `sender_0`.

Reply:
ah

Reply:
Coolio. Thanks for saving me a rabbit hole trip.